include/stl2/detail/memory/uninitialized_move.hpp

```diff
@@ -45,22 +44,33 @@ STL2_OPEN_NAMESPACE {
 	//
 	template <InputRange Rng, __NoThrowForwardIterator O>
 	requires
-		Constructible<value_type_t<O>, rvalue_reference_t<iterator_t<Rng>>> &&
-		__ReferenceTo<iterator_t<Rng>, value_type_t<O>>
+		Constructible<value_type_t<O>, rvalue_reference_t<iterator_t<Rng>>>
 	tagged_pair<tag::in(safe_iterator_t<Rng>), tag::out(O)>
 	uninitialized_move(Rng&& rng, O result)
 	{
 		return __stl2::uninitialized_move(
 			__stl2::begin(rng), __stl2::end(rng), std::move(result));
 	}
 
+	///////////////////////////////////////////////////////////////////////////
+	// uninitialized_move [Extension]
+	//
+	template <InputRange IRng, __NoThrowForwardRange ORng>
+	requires
+		Constructible<value_type_t<iterator_t<ORng>>, rvalue_reference_t<iterator_t<IRng>>>
+	tagged_pair<tag::in(safe_iterator_t<IRng>), tag::out(iterator_t<ORng>)>
+	uninitialized_move(IRng&& irng, ORng& orng)
```

Comment:
Likewise.

Reply:
Ditto "`ORng&&` and the return type `tagged_pair<tag::in(safe_iterator_t<IRng>), tag::out(safe_iterator_t<ORng>)>`"